lib/semantics/check-omp-clause.cc

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "check-omp-clause.h"
+#include "../parser/parse-tree.h"
+
+namespace Fortran::semantics {
+
+// 2.5 parallel-clause -> if-clause |
+//                        num-threads-clause |
+//                        default-clause |
+//                        private-clause |
+//                        firstprivate-clause |
+//                        shared-clause |
+//                        copyin-clause |
+//                        reduction-clause |
+//                        proc-bind-clause
+void OmpClauseChecker::Enter(const parser::OmpBlockDirective::Parallel &x) {
+  std::set<std::type_index> s{typeid(parser::OmpIfClause),
+      typeid(parser::OmpClause::NumThreads), typeid(parser::OmpDefaultClause),
+      typeid(parser::OmpClause::Private),
+      typeid(parser::OmpClause::Firstprivate),
+      typeid(parser::OmpClause::Shared), typeid(parser::OmpClause::Copyin),
+      typeid(parser::OmpReductionClause), typeid(parser::OmpProcBindClause)};
+  SetAllowed(s);
```

Comment:
This could be just `allowedClauses_ = {typeid(parser::OmpIfClause),` ...

Reply:
okay!